### ISSUE TITLE
Scalability

### ISSUE THREAD

**Issue by joglr** [git-truck/git-truck#595]
- Improve analyzation time with multi threading
 https://nodejs.org/api/worker_threads.html
 - Handle large JSON file output
   - Either skip cache for large projects or find better method for doing this
 - Improve visualization performance
   - State management
   - Coalescing

**Comment by dawidwoz** [comment]
As I'm new to the project, I want to ask if you think of splitting one big JSON file, as it is now, into a few small files and moving the analyse to a set of async tasks running in the background or on the demand, especially, the hydrate phase.

If so, was it rejected?

CC: @mircealungu

**Comment by mircealungu** [comment]
I remember @joglr mentioning the idea of splitting into multiple files. 
Analysis in async tasks - i'm not sure about that. Isn't the processing of the commits by definition a sequential process?

**Comment by joglr** [comment]
> Analysis in async tasks - i'm not sure about that. Isn't the processing of the commits by definition a sequential process? 

It used to be, but we switched to single git call for listing commits.

**Comment by joglr** [comment]
> I remember @joglr mentioning the idea of splitting into multiple files. 

Yeah we talked about that. Maybe an entirely different strategy could make sense for the cache, such as SQLite or something 🤷‍♂️

**Comment by mircealungu** [comment]
Hmm. SQLite is a very interesting idea that we should investigate!
